include/aspect/material_model/diffusion_dislocation.h

```diff
@@ -0,0 +1,200 @@
+/*
+  Copyright (C) 2011 - 2014 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef __aspect__model_diffusion_dislocation_h
+#define __aspect__model_diffusion_dislocation_h
+
+#include <aspect/material_model/interface.h>
+#include <aspect/simulator_access.h>
+#include <deal.II/base/parameter_handler.h>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+
+    /**
+     * A material model based on a viscous rheology including diffusion and
+     * dislocation creep.
+     *
+     * The effective viscosity is defined as the harmonic mean of the two
+     * effective viscosity functions describing diffusion and dislocation creep:
+     * \\[v_{eff} = \\left( \\frac{1}{v_{eff}^{diff}} + \\frac{1}{v_{eff}^{dis} \\right)^{-1}\\]
+     * where
+     * \\[v_{eff}^{diff} = A_{diff}^-1 \\exp\\left(\frac{E_{diff} + PV_{diff}}{RT}\\right)\\]
+     * \\[v_{eff}^{dis} =  A_{dis}^{\\frac{-1}{n_dis}} \\dot{\\varepsilon}^{\frac{1-n}{n}} \\exp\\left(\frac{E_{diff} + PV_{diff}}{n_{dis}RT}\\right)\\]
+     *
+     * where $\\dot{\\varepsilon}$ is the second invariant of the strain rate tensor,
+     *  $A_i$ are prefactors for each of the creep laws,
+     * $E_i$ are the activation energies, $V_i$ are the activation volumes,
+     * $\\rho_m$ is the mantle density, $R$ is the gas constant,
+     * $T$ is temperature, and $P$ is pressure.
```

Comment:
Can you say something like "Here, $i$ corresponds to either the diffusion or dislocation material parameters."